composer/trainer/trainer.py

```diff
@@ -49,10 +50,15 @@ class Trainer:
     Args:
         model (BaseMosaicModel): The model to train.
         train_dataloader_spec (DataloaderSpec): The dataloader spec for the training data.
-        eval_dataloader_spec (DataloaderSpec): The dataloader spec for the evaluation data.
+        eval_dataloader_spec (Optional[DataloaderSpec]): The dataloader spec for the evaluation
```

Comment:
Please make sure to rebase against dev. I don't think the diff has incorporated the latest dataloader changes.

Reply:
(I'll review the training loop changes after the rebase)